src/VisualStudio/Core/Def/Implementation/VirtualMemoryNotificationListener.cs

```diff
@@ -61,6 +68,17 @@ public int OnBroadcastMessage(uint msg, IntPtr wParam, IntPtr lParam)
                         }
 
                         _workspaceCacheService.FlushCaches();
+
+                        // turn off full solution analysis
+                        if (_workspace.Options.GetOption(RuntimeOptions.FullSolutionAnalysis))
+                        {
+                            _workspace.Services.GetService<IOptionService>().SetOptions(_workspace.Options.WithChangedOption(RuntimeOptions.FullSolutionAnalysis, false));
+
+                            // let user know full analysis is turned off due to memory concern
+                            // no close info bar action
+                            _workspace.Services.GetService<IErrorReportingService>().ShowErrorInfo(ServicesVSResources.FullSolutionAnalysisOff, () => { });
```

Comment:
No no.  I understood the point you're making here.  What i'm saying is that the UI does not convey that properly at all.

You're telling users in the UI "if you want this back, here's how you go turn it on".  There's nothing to tell them "but if you do this, you'll likely OOM".

So since you've already decided you're going to tell them how to do it, you might as well give them the way to do it.  Right now the approach you're taking makes for a *very* user-unfriendly UI.  Telling a user that they need to go into tools|options and then *search* for something, is a *super* unpleasant set of instructions.  It's antithetical to how computers should behave.  

--

We have to decide on a few things:

Do we even want to allow users to re-enable this?  If no, then we don't need a button, and we don't need all that text explaining how to turn it back on.  If yes, then we should go for the much better UI approach.  

If we're worried about them causing an OOM by doing this, then we shouldn't bother giving them the button *or* telling them how to re-enable it.  Right now we've got a really unpleasant approach.  We don't want them to re-enable, unless they're on another solution, but we want htem to have to go through a really unpleasant set of steps to re-enable in that case.  That's not a good user experience *at all*.